relayer/app/cursor/stream.go

```diff
@@ -0,0 +1,137 @@
+package cursor
+
+import (
+	"context"
+	"math"
+	"slices"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+)
+
+type stream struct {
+	SrcVersion xchain.ChainVersion
+	DestChain  netconf.Chain
+	Network    netconf.Network
```

Comment:
Network is a complex thing, not sure it should be part of "stream". Also, it isn't used...